File: apps/BuddyPlay/shared/BuddyCore/Sources/BuddyCore/Storage/DeviceIdProvider.swift
import Foundation

/// `UserDefaults`-backed stable device UUID. Generated once on first
/// access; rotatable from Settings → Reset device ID.
public final class DeviceIdProvider {

    public static let defaultsKey = "buddyplay.deviceId"

    private let defaults: UserDefaults

    public init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    public func deviceId() -> UUID {
        if let str = defaults.string(forKey: Self.defaultsKey),
           let uuid = UUID(uuidString: str) {
            return uuid
        }
        let new = UUID()
        defaults.set(new.uuidString, forKey: Self.defaultsKey)
        return new
    }

    public func reset() -> UUID {
        let new = UUID()
        defaults.set(new.uuidString, forKey: Self.defaultsKey)
        return new
    }
}

File: apps/BuddyPlay/android/core/src/main/java/com/americangroupllc/buddyplay/core/storage/DeviceIdProvider.kt
package com.americangroupllc.buddyplay.core.storage

import java.util.UUID

/**
 * Stable device UUID. Generated once on first access; rotatable from
 * Settings → Reset device ID.
 *
 * `:core` declares the interface so domain code can depend on it; `:app`
 * provides the SharedPreferences-backed implementation.
 */
interface DeviceIdProvider {
    fun deviceId(): String
    fun reset(): String
}

/** In-memory test/JVM implementation. */
class InMemoryDeviceIdProvider(initial: String = UUID.randomUUID().toString()) : DeviceIdProvider {
    private var id: String = initial
    override fun deviceId(): String = id
    override fun reset(): String { id = UUID.randomUUID().toString(); return id }
}
